pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -911,3 +912,203 @@ func removeNodeIPAddressesFromBackendPool(
 
 	return changed
 }
+
+type backendPoolTypePodIP struct {
+	*Cloud
+}
+
+func newBackendPoolTypePodIP(c *Cloud) BackendPool {
+	return &backendPoolTypePodIP{c}
+}
+
+func (bpi *backendPoolTypePodIP) CleanupVMSetFromBackendPoolByCondition(_ *network.LoadBalancer, _ *v1.Service, _ []*v1.Node, _ string, _ func(string) bool) (*network.LoadBalancer, error) {
+	return nil, errors.New("CleanupVMSetFromBackendPoolByCondition is not implemented for pod IP backend pool")
+}
+
+func (bpi *backendPoolTypePodIP) EnsureHostsInPool(service *v1.Service, _ []*v1.Node, _, _, clusterName, lbName string, backendPool network.BackendAddressPool) error {
+	isIPv6 := isBackendPoolIPv6(ptr.Deref(backendPool.Name, ""))
+
+	var (
+		changed               bool
+		err                   error
+		podIPsToBeAdded []string
+		endpointsliceName string
+		endpointSliceNames [] string
+		numOfAdd int
+	)
+
+	endpointSliceList, err := bpi.getEndpointSliceListForService(service)
+
+	if err != nil {
+		klog.Errorf("bpi.EnsureHostsInPool: failed to get endpoint slice list for service %q, error: %s", service.Name, err.Error())
+		return err
+	}
+
+	lbBackendPoolName := bpi.getBackendPoolNameForService(service, clusterName, isIPv6)
+
+	/* Remove all addresses from the backend pool and add the addresses from all the 
+	   endpoint-slices pertaining to a service.*/
+	if strings.EqualFold(ptr.Deref(backendPool.Name, ""), lbBackendPoolName) &&
+		backendPool.BackendAddressPoolPropertiesFormat != nil {
+		if backendPool.LoadBalancerBackendAddresses == nil {
+			lbBackendPoolAddresses := make([]network.LoadBalancerBackendAddress, 0)
+			backendPool.LoadBalancerBackendAddresses = &lbBackendPoolAddresses
+		} else {
+			removeNodeIPAddressesFromBackendPool(backendPool, []string{}, true, false)
+		}
+
+		for _, ES := range endpointSliceList {
+
+			if ES.AddressType == discovery_v1.AddressTypeIPv6 && !isIPv6 {
+				continue
+			}
+
+			if ES.AddressType == discovery_v1.AddressTypeIPv4 && isIPv6 {
+				continue
+			}
+
+			for _, endpoint := range ES.Endpoints {
+
+				if endpoint.Conditions.Ready != nil && !*endpoint.Conditions.Ready {
+					// Skip endpoints that are not ready
+					continue
+				}
+
+				for _, address := range endpoint.Addresses {
+					klog.V(6).Infof("bpi.EnsureHostsInPool: adding ip address %s", address)
+					podIPsToBeAdded = append(podIPsToBeAdded, address)
+					endpointsliceName = strings.ToLower(fmt.Sprintf("%s/%s", ES.Namespace, ES.Name))
+					endpointSliceNames = append(endpointSliceNames,endpointsliceName)
```

Comment:
Should this be added for every address?

Reply:
A new property is used in the Address for the slice/ipgroup name. I think we don't need to populate this field if we the updateBackendIpGroup API.